blockchain/v1/reactor.go

```diff
@@ -67,6 +68,9 @@ type BlockchainReactor struct {
 	eventsFromFSMCh chan bcFsmMessage
 
 	swReporter *behaviour.SwitchReporter
+
+	// Atomic integer (0 - sync in progress, 1 - finished syncing)
+	syncEnded int32
```

Comment:
why not just store a reference here?

Reply:
what are the benefits?

Reply:
You don't need to do `&(bcR.syncEnded)`

Reply:
But then I will need to set initial value of 0 in constructor. I prefer my version for that reason.

Reply:
but zero is the default value. What would you explicitly need to set it to?

Reply:
This isn't blocking, but given that you continuously need to dereference, I think it makes it harder to read. Up to you :)

Reply:
> but zero is the default value.

not if it's a pointer! nil is zero value for pointer

Reply:
No, I meant it's zero if you make it an `int32` (per my suggestion). So my question is, is zero a (sync in progress) the correct default value?

Reply:
yes, 0 is correct default value

Reply:
> But then I will need to set initial value of 0 in constructor.

Then you don't need to do this.